src/Contract/Test/Utils.purs

```diff
@@ -629,3 +632,14 @@ checkTxHasMetadata
   -> Contract r Boolean
 checkTxHasMetadata txHash =
   mkCheckFromAssertion <<< assertTxHasMetadataImpl mempty txHash
+
+--------------------------------------------------------------------------------
+-- function to cancel aff fibers on signal
+--------------------------------------------------------------------------------
+
+foreign import exitCode :: Int -> Effect Unit
+
+interruptOnSignal :: forall a. Signal -> Fiber a -> Effect Unit
+interruptOnSignal signal fiber = Process.onSignal signal do
+  launchAff_ do
```

Comment:
Can we fork here though? I assumed signal handlers must be sync, or the default action (e.g. exiting) will be executed before the async action has a chance to run. Is that not true?

Reply:
not sure, feel still a little bit lost with Aff/Effect.. But as I understand it the default action (not exiting, but removing the temp dirs) is supposed to run before the async action which is killing the fibre

Reply:
Indeed, SIGINT does not exit once you add a custom handler. So, instead of calling exit, we cancel the fiber which has the same effect in the end: the process will exit when there are no more possible events, which eventually happens once the fiber has run its cleanup.